dcalc/DmpCeff.cc

```diff
@@ -1766,30 +1766,32 @@ DmpError::DmpError(const char *what) :
   //printf("DmpError %s\n", what);
 }
 
-// This saves about 2.5% in overall run time on designs with SPEF.
-// https://codingforspeed.com/using-faster-exponential-approximation
+// Fast approximate exponential. See
+// https://www.schraudolph.org/pubs/Schraudolph99.pdf
 static double
 exp2(double x)
 {
-  if (x < -12.0)
-    // exp(-12) = 6.1e-6
+  // For arguments greater than 707.703272 Shraudolphs approximation degrades severely in accuracy.
```

Comment:
I don't see a check for this in the code.

Reply:
Maybe you looked at an older version? The check below ensures that we return +Inf for x > 707.703272 and zero for x < -707.703272.

Reply:
FWIW: the old implementation exceeds a 3% relative error already at x=15.5 and then things go bad quickly. :-)

Reply:
Just to illustrate it graphically: 
![image](https://github.com/The-OpenROAD-Project/OpenSTA/assets/16907534/cd449c14-4bfd-4492-bab8-4c9bec23e8c9)

Reply:
88 happens to be the points where single precision overflows, so I suspect that somebody took the constant `4096` from a piece of code that was made for computing `exp<float>(x)`. Probably doesn't matter here, buty just thought I'd mention it for the record.

Reply:
@tspyrou I'm used to work on precisely specified numerical functions (in SW or HW), so I thought I'd try to bring this up to a similar standard. We can remove the comment if you prefer.

Reply:
I must have been looking at an older version.